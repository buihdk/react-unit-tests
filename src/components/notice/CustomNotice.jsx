import React from 'react';
import PropTypes from 'prop-types';
import { Button } from 'antd';
import I18n from 'i18n-js';

import { noop } from 'src/utils';

const styles = {};

const CustomNotice = ({ noticeQuery, actionText, onClick }) => (
  <div className={styles.container}>
    <span className={styles.noticeText}>
      {I18n.t('common.noticeText.part1')}
      <strong> {noticeQuery} </strong>
      {I18n.t('common.noticeText.part2')}
    </span>
    <Button type="ghost" className={styles.actionText} onClick={onClick}>
      {actionText}
    </Button>
  </div>
);

CustomNotice.propTypes = {
  noticeQuery: PropTypes.string,
  actionText: PropTypes.string,
  onClick: PropTypes.func,
};

CustomNotice.defaultProps = {
  noticeQuery: '',
  actionText: '',
  onClick: noop,
};

export default CustomNotice;
