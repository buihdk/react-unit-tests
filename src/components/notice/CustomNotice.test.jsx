import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import CustomNotice from './CustomNotice';

describe('CustomNotice', () => {
  test('renders without crashing', () => {
    const wrapper = shallow(<CustomNotice />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
