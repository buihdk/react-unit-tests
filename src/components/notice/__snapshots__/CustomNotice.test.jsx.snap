// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomNotice renders without crashing 1`] = `
<div>
  <span>
    Your query
    <strong>
       
       
    </strong>
    is in invalid format and not supported in Campaign UI.
  </span>
  <Button
    block={false}
    ghost={false}
    htmlType="button"
    loading={false}
    onClick={[Function]}
    type="ghost"
  />
</div>
`;
